### PR TITLE
Start new Cans Assessment CANS-176

### PR DESCRIPTION
**Pull request by jonbwalker**
* add add-cans button to assessment form header
* fetch a newly created assessment form on add-cans click
* add client name to top of assessment form
* add client assessment history card on client page
* update assessment route to accept optional id

## Description
This feature adds the Add-Cans button to the client page at the top of the Assessment History form
Clicking the Add-Cans button will take you to /assessments where it will display a new assessment form to be filled out. 

This feature also adds the child's name to the top of the assessment form.

## Tests

- [ ] I used TDD to develop the feature
- [x] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I ran all my tests locally which were green.
- [ ] My code adds no new issues to Code Climate
- [ ] I ran all the linters for the project.
- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [ ] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

**Comment by jonbwalker** [comment]
@abelayt @edgibbs @denys-davydov This PR includes the complete feature, WIP was added to because I'm finishing up Assessment.test, it's still in progress. All other files can be reviewed

I'll remove WIP once Assessment.test is complete

File: client/src/components/Client/ClientAssessmentHistory.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          clientLastName: clientLastName,
+        }}
+        className={'add-cans-button'}
+      >
+        <Button
+          size="small"
+          color="inherit"
+          classes={{ label: 'add-cans-button' }}
+        >
+          Add Cans
```
Comment (edgibbs) [inline client/src/components/Client/ClientAssessmentHistory.js]: The styling on the hover effect of this button looks off.

File: client/src/components/Assessment/Assessment.js [hunk 2]
```diff
@@ -1,5 +1,11 @@
 import React, { Component, Fragment } from 'react';
-import { AssessmentFormHeader, Domain, DomainsGroup, AssessmentService, I18nService } from './';
+import {
```
Comment (edgibbs) [inline client/src/components/Assessment/Assessment.js]: We probably need to do the full prettier run across the codebase soon.

File: client/src/components/Assessment/Assessment.js [hunk 3]
```diff
@@ -135,13 +159,18 @@ class Assessment extends Component {
   };
 
   render = () => {
-    const assessmentState = this.state.assessment.state || {};
+    const { clientFirstName, clientLastName } = this.props.location;
```
Comment (edgibbs) [inline client/src/components/Assessment/Assessment.js]: I see how this hooks together, but why is clientFirstName, clientLastName considered part of location?

File: client/src/components/Assessment/Assessment.service.js [hunk 4]
```diff
@@ -5,7 +5,12 @@ export class AssessmentService {
     return appApi.get(`/assessments/${id}`).then(response => response.data);
   }
   static update(id, assessment) {
-    return appApi.put(`/assessments/${id}`, assessment).then(response => response.data);
+    return appApi
+      .put(`/assessments/${id}`, assessment)
+      .then(response => response.data);
+  }
+  static fetchNewAssessment() {
+    return appApi.get('/instruments/1').then(response => response.data);
```
Comment (edgibbs) [inline client/src/components/Assessment/Assessment.service.js]: Is this just hardcoded for now as 1?

File: client/src/components/Assessment/Assessment.test.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { Assessment, AssessmentService } from './index';
+import Domain from './Domain';
+
+describe('<Assessment />', () => {
```
Comment (edgibbs) [inline client/src/components/Assessment/Assessment.test.js]: So I'm big on using a context style here, so I'd structure these tests the following way:

```javascript
describe('with no existing assessment', () => {
  it('calls fetchNewAssessment', () => {
  });
});
describe('with an existing assessment', () => {
  it('calls fetchAssessment', () => {
  });
});
```

File: client/src/components/Assessment/Assessment.test.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      Assessment.prototype,
+      'fetchAssessment'
+    );
+    const wrapper = shallow(<Assessment {...props} />);
+    wrapper.instance().componentDidMount();
+    expect(fetchAssessmentSpy).toHaveBeenCalled();
+  });
+});
+
+describe('fetchNewAssessment', async () => {
```
Comment (edgibbs) [inline client/src/components/Assessment/Assessment.test.js]: where's the await, may be getting lucky with the timing here.

File: client/src/components/Assessment/AssessmentFormHeader.js [hunk 7]
```diff
@@ -4,12 +4,23 @@ import { Row, Col, Label, Input } from 'reactstrap';
 import './style.css';
 
 class AssessmentFormHeader extends Component {
+  renderClientName() {
+    const { clientFirstName, clientLastName } = this.props;
+    return (
+      <Fragment>
+        {clientFirstName && clientLastName ? (
```
Comment (edgibbs) [inline client/src/components/Assessment/AssessmentFormHeader.js]: This ternary probably wants a friendly neighborhood test.

File: client/src/components/Client/style.sass [hunk 8]
```diff
@@ -27,4 +27,14 @@ a:visited
...
   text-decoration: underline
   margin: 4px 4px 4px 8px
-  
\ No newline at end of file
+
+//**************************
+//  ClientAssessmentHistory
+//**************************
+.assessment-history-container
+    margin-top: -6 !important
```
Comment (edgibbs) [inline client/src/components/Client/style.sass]: important! seems dangerous here, can we avoid it?

File: client/src/routes/Routes.js [hunk 9]
```diff
@@ -8,7 +8,7 @@ class Routes extends Component {
     return (
       <Router>
         <Switch>
-          <Route path="/assessments" component={Assessment} />
+          <Route path="/assessments/:id?" component={Assessment} />
```
Comment (edgibbs) [inline client/src/routes/Routes.js]: Feels like it needs an additional test where it does get the optional id.

File: client/src/components/Assessment/Assessment.js (hunk 3 above)
Comment (jonbwalker) [reply client/src/components/Assessment/Assessment.js]: location is object used in react-router for passing additional data as your routing from one component to another. The naming isn't very intuitive.

File: client/src/components/Assessment/Assessment.service.js (hunk 4 above)
Comment (jonbwalker) [reply client/src/components/Assessment/Assessment.service.js]: @denys-davydov can you explain how we will handle this in the future?

File: client/src/routes/Routes.js (hunk 9 above)
Comment (jonbwalker) [reply client/src/routes/Routes.js]: Good call, I'll add one.

File: client/src/components/Assessment/Assessment.js (hunk 3 above)
Comment (edgibbs) [reply client/src/components/Assessment/Assessment.js]: OK, cool then.

File: client/src/components/Assessment/AssessmentFormHeader.js (hunk 7 above)
Comment (jonbwalker) [reply client/src/components/Assessment/AssessmentFormHeader.js]: done! 
deefe80

Comment (jonbwalker) [reply client/src/components/Assessment/AssessmentFormHeader.js]: I also ran prettier on AssessmentFormHeader, so you'll see some changes there too.

File: client/src/components/Assessment/Assessment.test.js (hunk 5 above)
Comment (jonbwalker) [reply client/src/components/Assessment/Assessment.test.js]: I agree, this reads much better: 836c901

File: client/src/components/Assessment/Assessment.test.js (hunk 6 above)
Comment (jonbwalker) [reply client/src/components/Assessment/Assessment.test.js]: I removed this test for now, see me when you have a free minute.

File: client/src/components/Client/ClientAssessmentHistory.js (hunk 1 above)
Comment (edgibbs) [reply client/src/components/Client/ClientAssessmentHistory.js]: cool, button looks good.

File: client/src/components/Assessment/Assessment.service.js (hunk 4 above)
Comment (denys-davydov) [reply client/src/components/Assessment/Assessment.service.js]: For now, we have only 1 instrument, so we can just use hardcoded id. Eventually, we will have to fetch instrument by client's county or by a different property, will know later.

**Comment by edgibbs** [comment]
All the codeclimate issues fine false positive duplicate code issues.

**Pull request merged**